src/core/thread/mle_router.cpp

```diff
@@ -2962,7 +2961,10 @@ Error MleRouter::SendChildUpdateRequest(Child &aChild)
     if (!aChild.IsStateValid())
     {
         SuccessOrExit(error = message->AppendTlvRequestTlv(kTlvs));
-        aChild.GenerateChallenge();
+        if (!aChild.IsStateRestored())
+        {
```

Comment:
May also help adding a short comment here. Folks looking a the code will wonder why the `GenerateChallenge()` is being skipped under `IsStateRestored()`.

Reply:
thanks, incorporated your comments to the code snippet and and the commit log.